http/codegen/openapi/v3/types.go

```diff
@@ -440,6 +440,87 @@ func hashAttribute(att *expr.AttributeExpr, h hash.Hash64, seen map[string]*uint
 		*res = hashString(t.Name(), h)
 	}
 
+	// Validations can change the type of an attribute: just because two things
+	// are strings, if they have different validation rules, we cannot treat them
+	// as the same type.
+	hv := hashValidation(att.Validation, h)
+	if hv != 0 {
+		*res = orderedHash(*res, hv, h)
+	}
+
+	return res
+}
+
+func hashValidation(val *expr.ValidationExpr, h hash.Hash64) uint64 {
+	h.Reset()
+	if val == nil {
+		return 0
+	}
+	var parts []uint64
+	if val.Format != "" {
+		parts = append(parts, hashString(string(val.Format), h))
+	}
+	if val.Pattern != "" {
+		parts = append(parts, hashString(val.Pattern, h))
+	}
+	if val.MinLength != nil {
```

Comment:
This algo may require a bit of tweaking, as it stands the following two validations would result in the same hash:
```go
Pattern("1")
```
and 
```
MinLength(1)
```
Obviously there are many other combinations that would result in the same hash being generated. Maybe we need to use something like https://github.com/gohugoio/hashstructure,

Might also be good to add the example above (and maybe others) to tests.